Update Tags, Map, and app layout components to use sx prop for styles

tillyw opened the pull request:
## Associated issues
closes https://github.com/cityofaustin/atd-data-tech/issues/23660

## Testing
**URL to test:** 
<!---
deploy-preview-[pr-number]--atd-moped-main.netlify.app/moped/
--->
https://deploy-preview-1634--atd-moped-main.netlify.app/

**Steps to test:**

Compare the following views in the deploy preview to staging

- Project summary tags section
<img width="734" height="399" alt="Screenshot 2025-07-31 at 4 24 22 PM" src="https://github.com/user-attachments/assets/c55c631a-cbd0-4d76-b83d-1fdad74c7a42" />

- Project summary map fallback state (must select a project with no components)
<img width="762" height="481" alt="Screenshot 2025-07-31 at 4 56 58 PM" src="https://github.com/user-attachments/assets/864eb1b7-ef86-4c17-a764-3daf687eae68" />

- Projects list view map view
Note: this component was already refactored by @mateoclarke in a previous pr
<img width="1878" height="659" alt="Screenshot 2025-07-31 at 4 59 57 PM" src="https://github.com/user-attachments/assets/f8de9c24-7857-404a-81e4-6c426f5f17ac" />

- Component map toolbar
<img width="1920" height="72" alt="Screenshot 2025-07-31 at 5 02 59 PM" src="https://github.com/user-attachments/assets/e3fcd59f-acb2-4535-a1f6-686a0c4d4d8e" />

- App footer
<img width="1920" height="101" alt="Screenshot 2025-07-31 at 5 07 18 PM" src="https://github.com/user-attachments/assets/8c2a3f2c-9c2f-4f1a-b21c-9fdbc4665e3b" />

- App top bar
<img width="1920" height="124" alt="Screenshot 2025-07-31 at 5 08 21 PM" src="https://github.com/user-attachments/assets/91102abc-97b1-4722-9c6a-065c1d66beca" />

- App dropdown menu (mobile view only)
<img width="352" height="760" alt="Screenshot 2025-07-31 at 5 10 51 PM" src="https://github.com/user-attachments/assets/0b653dc5-e072-4bf8-bc4d-e97fa52255f2" />

---
#### Ship list
- [x] Code reviewed 
- [x] Product manager approved
- [x] Product manager checked [DB view dependencies](https://atd-dts.gitbook.io/moped-documentation/product-management/updating-the-arcgis-online-components-database-view)
- ~[ ] Product manager added to [QA test script](https://docs.google.com/spreadsheets/d/1n_O6MLh9cwwPf57HUM394Ea-z9uuoEV1-QW4axNZXLE/edit#) if applicable~

tillyw commented:
actually please hold off on review, I need to fix an issue with the selected tab active state on the main navigation bar.

tillyw commented on moped-editor/src/layouts/DashboardLayout/TopBar.js:
```diff
@@ -1,16 +1,13 @@
 import React, { useState } from "react";
 import { Link as RouterLink } from "react-router-dom";
-import clsx from "clsx";
-import PropTypes from "prop-types";
```
I believe that, once we dispense of all uses of `makeStyles`, we'll be able to uninstall both of these libraries as well!

tillyw commented on moped-editor/src/layouts/DashboardLayout/TopBar.js:
```diff
@@ -1,16 +1,13 @@
 import DropdownMenu from "src/layouts/DashboardLayout/NavBar/DropdownMenu";
 import NavigationSearchInput from "src/layouts/DashboardLayout/NavBar/NavigationSearchInput";
-import NavLink from "src/components/NavLink";
+import { NavLink } from "react-router-dom";
```
I believe we will also be able to delete this custom component that was made to support some deprecated behavior in `react-router-dom` and be able to just use the standard component. The custom component is currently used in a dozen or so files.

tillyw replied:
you're right, thanks for pointing that out! glad we can delete this custom component :)

tillyw commented:
thanks @roseeichelmann for catching these discrepancies and @mddilley for sharing your discoveries and workarounds! I'll get these integrated and make sure to use our existing spacing and theme values where possible

chiaberry commented:
@roseeichelmann I think the tags are still not aligned in the deploy preview because the deploy preview failed to build. I am running locally and the Tags title is left aligned

chiaberry commented on moped-editor/src/views/projects/projectView/ProjectSummary/TagsSection.js:
```diff
@@ -160,7 +134,24 @@ const TagsSection = ({ projectId, handleSnackbar }) => {
+              {
+                theme.spacing(1);
+              }
+              {
+                theme.spacing(2);
+              }
```
here is what is blocking the build

tillyw replied:
my apologies, y'all! i fixed the build and made some tweaks so the the tags section aligns perfectly with the subproject section both vertically and horizontally. there are some add values but it works!

<img width="522" height="411" alt="Screenshot 2025-08-12 at 3 54 42 PM" src="https://github.com/user-attachments/assets/7bde74d7-9a23-44b9-be87-3ad2e5f981cf" />

tillyw commented:
@mddilley i made the changes you requested! replaced a number of pixel values with theme shorthand and removed a couple of theme imports. ready for a final look!
